core/src/main/java/org/apache/calcite/rel/externalize/RelEnumTypes.java

```diff
@@ -105,6 +107,6 @@ public static String fromEnum(Enum enumValue) {
   /** Converts a string to an enum value.
    * The converse of {@link #fromEnum(Enum)}. */
   static <E extends Enum<E>> E toEnum(String name) {
-    return (E) ENUM_BY_NAME.get(name);
+    return (E) Objects.requireNonNull(ENUM_BY_NAME.get(name));
```

Comment:
This seems to be a change in the method contract. Previously the method would return null, now it throws. At least change the javadoc to make the contract explicit.